### PR TITLE
Upgrade to Hibernate 5.4.25.Final

### PR DESCRIPTION
**Pull request by VinodAnandan**
Closes gh-24331

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@VinodAnandan Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=24332)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=24332) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@VinodAnandan Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=24332)!

**Review by m1ngyuan — COMMENTED** [review]
The latest version should be 5.2.25.Final

File: spring-boot-project/spring-boot-dependencies/build.gradle [hunk 1]
```diff
@@ -398,7 +398,7 @@ bom {
 			]
 		}
 	}
-	library("Hibernate", "5.4.23.Final") {
+	library("Hibernate", "5.4.24.Final") {
```
Comment (m1ngyuan) [inline spring-boot-project/spring-boot-dependencies/build.gradle:401]: The latest version should be 5.2.25.Final

Comment (VinodAnandan) [reply spring-boot-project/spring-boot-dependencies/build.gradle]: Thanks for the review. I have made the change.

**Comment by philwebb** [comment]
Thanks for the PR, but as mentioned in the [pull request template](https://raw.githubusercontent.com/spring-projects/spring-boot/master/.github/PULL_REQUEST_TEMPLATE.md) we have a semi-automated process for dependency upgrades that we prefer to use.

**Comment by iwt-gregorpoloczek** [comment]
@philwebb I'm having trouble to understand what finally happend to this PR and the other https://github.com/spring-projects/spring-boot/issues/24331 you makred as superseded. Do you know if this dependency upgrade will be available in the next spring boot version slated for 10th of december? Thanks!

**Comment by snicoll** [comment]
@iwt-gregorpoloczek Phil mentioned we have a semi-automated process for dependency upgrades. We will run that prior to any release as we always do.

**Comment by iwt-gregorpoloczek** [comment]
@snicoll I wasn't aware, that this process is always is run. Thanks for the info!

**Pull request closed without merging**